src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -122,4 +130,97 @@ public Stats deserialize(JsonElement jsonElement, Type type,
             }
         }
     }
+
+    /**
+     * Deserializer for Arguments class.
+     * Breaks the incoming arguments into args and kwargs parts
+     * and fills a new Arguments instance.
+     */
+    private class ArgumentsDeserializer implements JsonDeserializer<Arguments> {
+
+        private static final String KWARG_KEY = "__kwarg__";
+
+        @Override
+        public Arguments deserialize(JsonElement json, Type typeOfT,
+                JsonDeserializationContext context) throws JsonParseException {
+            Arguments result = new Arguments();
+
+            if (json != null && json.isJsonArray()) {
+                for (JsonElement jsonElement : json.getAsJsonArray()) {
+                    fillArgs(result, jsonElement);
+                }
+            }
+
+            return result;
+        }
+
+        /**
+         * Fills args/kwargs to given Arguments instance based on JSON data
+         * from jsonElement.
+         *
+         * @param result Arguments to be filled
+         * @param jsonElement input JSON data
+         */
+        private void fillArgs(Arguments result, JsonElement jsonElement) {
+            boolean isKwarg = isKwarg(jsonElement);
+            // we don't need __kwarg__ attribute:
+            // args/kwargs are stored to separate attributes of Arguments class
+            // so we don't need this information anymore
```

Comment:
@hustodemon do you mean [the pop call here](https://github.com/saltstack/salt/blob/develop/salt/minion.py#L264)?

In that case it is discarded, yes, but as an input to Salt. We are talking about a deserializer here, so we are parsing something from Salt.

How about moving line 166 to line 169, so that we are definitely sure we are not discarding anything useful? Would that work?